inspirehep/dojson/hepnames/fields/bd1xx.py

```diff
@@ -393,36 +393,37 @@ def experiments2marc(self, key, value):
     }
 
 
-@hepnames.over('phd_advisors', '^701..')
+@hepnames.over('advisors', '^701..')
 @utils.for_each_value
-@utils.filter_values
-def phd_advisors(self, key, value):
-    degree_type_map = {
-        "phd": "PhD",
-        "master": "Master"
+def advisors(self, key, value):
+    DEGREE_TYPES_MAP = {
+        'Bachelor': 'Bachelor',
+        'UG': 'Bachelor',
+        'MAS': 'Master',
+        'master': 'Master',
+        'Master': 'Master',
+        'PhD': 'PhD',
+        'PHD': 'PhD',
     }
-    degree_type = None
-    if value.get("g"):
-        degree_type_raw = force_force_list(value.get('g'))[0]
-        degree_type = degree_type_map.get(
-            degree_type_raw.lower(),
-            degree_type_raw
-        )
+
+    _degree_type = force_single_element(value.get('g'))
+    degree_type = DEGREE_TYPES_MAP.get(_degree_type, 'Other')
+
     return {
-        'id': value.get("i"),
-        'name': value.get("a"),
-        'degree_type': degree_type
+        'id': value.get('i'),
```

Comment:
Not part of the schema. This actually needs the terrible identifier problem to be solved.